src/download/web.ts

```diff
@@ -0,0 +1,114 @@
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @param [customOptions.endpointPath="/"] - The relative URL path of the portal endpoint to contact.
+ * @returns - The full URL that was used.
+ * @throws - Will throw if the skylinkUrl does not contain a skylink or if the path option is not a string.
+ */
+export function downloadFile(this: SkynetClient, skylinkUrl: string, customOptions?: CustomDownloadOptions): string {
+  /* istanbul ignore next */
+  if (typeof skylinkUrl !== "string") {
+    throw new Error(`Expected parameter skylinkUrl to be type string, was type ${typeof skylinkUrl}`);
+  }
+
+  const opts = { ...defaultDownloadOptions, ...this.customOptions, ...customOptions, download: true };
+  const url = this.getSkylinkUrl(skylinkUrl, opts);
+
+  // Download the url.
+  window.location.assign(url);
+
+  return url;
+}
+
+/**
+ * Initiates a download of the content of the skylink at the Handshake domain.
+ *
+ * @param this - SkynetClient
+ * @param domain - Handshake domain.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @param [customOptions.endpointPath="/hns"] - The relative URL path of the portal endpoint to contact.
+ * @returns - The full URL that was used.
+ * @throws - Will throw if the input domain is not a string.
+ */
+export async function downloadFileHns(
+  this: SkynetClient,
+  domain: string,
+  customOptions?: CustomDownloadOptions
+): Promise<string> {
+  /* istanbul ignore next */
+  if (typeof domain !== "string") {
+    throw new Error(`Expected parameter domain to be type string, was type ${typeof domain}`);
+  }
+
+  const opts = { ...defaultDownloadHnsOptions, ...this.customOptions, ...customOptions, download: true };
+  const url = this.getHnsUrl(domain, opts);
+
+  // Download the url.
+  window.location.assign(url);
+
+  return url;
+}
+
+/**
+ * Opens the content of the skylink within the browser.
+ *
+ * @param this - SkynetClient
+ * @param skylinkUrl - Skylink string. See `downloadFile`.
+ * @param [customOptions] - Additional settings that can optionally be set. See `downloadFile` for the full list.
+ * @param [customOptions.endpointPath="/"] - The relative URL path of the portal endpoint to contact.
+ * @returns - The full URL that was used.
+ * @throws - Will throw if the skylinkUrl does not contain a skylink or if the path option is not a string.
+ */
+export function openFile(this: SkynetClient, skylinkUrl: string, customOptions?: CustomDownloadOptions): string {
+  /* istanbul ignore next */
+  if (typeof skylinkUrl !== "string") {
+    throw new Error(`Expected parameter skylinkUrl to be type string, was type ${typeof skylinkUrl}`);
+  }
+
+  const opts = { ...defaultDownloadOptions, ...this.customOptions, ...customOptions };
+  const url = this.getSkylinkUrl(skylinkUrl, opts);
+
+  window.open(url, "_blank");
```

Comment:
I think we might need to add noopener and noreferrer to this so it's safer

https://web.dev/external-anchors-use-rel-noopener/

https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noreferrer
https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noopener